workers/unity/Packages/com.improbable.gdk.core/Worker/ConnectionHandlers/MockConnectionHandler.cs

```diff
@@ -0,0 +1,90 @@
+using Improbable.Worker.CInterop;
+
+namespace Improbable.Gdk.Core
+{
+    public class MockConnectionHandler : IConnectionHandler
+    {
+        private uint updateId;
+
+        private readonly ViewDiff diff = new ViewDiff();
+
+        public void ClearDiff()
+        {
+            diff.Clear();
+        }
+
+        public void CreateEntity(long entityId, EntityTemplate template)
+        {
+            var handler = new EntityTemplateDynamicHandler(template, entityId, diff);
+            DynamicConverter.ForEachComponent(handler);
+        }
+
+        public void ChangeAuthority(long entityId, uint componentId, Authority newAuthority)
+        {
+            diff.SetAuthority(entityId, componentId, newAuthority);
+        }
+
+        public void UpdateComponent<T>(long entityId, uint componentId, T update) where T : ISpatialComponentUpdate
+        {
+            diff.AddComponentUpdate(update, entityId, componentId, updateId++);
```

Comment:
This means we can't have updates and events arriving atomically

Reply:
Can add a method to do that